controllers/grid/queries/form/QueryNoteForm.inc.php

```diff
@@ -141,24 +145,39 @@ public function execute(...$functionArgs)
         $query = $this->getQuery();
         if ($query->getIsClosed()) {
             $headNote = $query->getHeadNote();
-            if ($user->getId() != $headNote->getUserId()) {
+            if ($currentUser->getId() != $headNote->getUserId()) {
                 // Re-open the query.
                 $query->setIsClosed(false);
                 $queryDao = DAORegistry::getDAO('QueryDAO'); /** @var QueryDAO $queryDao */
                 $queryDao->updateObject($query);
             }
         }
 
+        $submissionId = $query->getAssocId();
+        $submission = Repo::submission()->get($submissionId);
+
+        $mailable = new MailDiscussionMessage($request->getContext(), $submission, $query);
+        $emailTemplate = Repo::emailTemplate()->getByKey($request->getContext()->getId(), $mailable::getEmailTemplateKey());
+        $context = $request->getContext();
+        $mailable->body($emailTemplate->getLocalizedData('body'))
+                ->subject('[' . $context->getLocalizedAcronym() . '] '."Re: ".$emailTemplate->getLocalizedData('subject'))
+                ->sender($currentUser);
+
         $notificationDao = DAORegistry::getDAO('NotificationDAO'); /** @var NotificationDAO $notificationDao */
         $queryDao = DAORegistry::getDAO('QueryDAO'); /** @var QueryDAO $queryDao */
 
         // Always include current user to query participants
-        if (!in_array($user->getId(), $queryDao->getParticipantIds($query->getId()))) {
-            $queryDao->insertParticipant($query->getId(), $user->getId());
+        if (!in_array($currentUser->getId(), $queryDao->getParticipantIds($query->getId()))) {
+            $queryDao->insertParticipant($query->getId(), $currentUser->getId());
         }
 
         $notificationManager = new NotificationManager();
+        /** @var NotificationSubscriptionSettingsDAO $notificationSubscriptionSettingsDAO */
+        $notificationSubscriptionSettingsDao = DAORegistry::getDAO('NotificationSubscriptionSettingsDAO');
+
         foreach ($queryDao->getParticipantIds($query->getId()) as $userId) {
+            $user = Repo::user()->get((int) $userId);
```

Comment:
It looks like `$user` isn't used until the end of the loop, in `$mailable->recipients([$user])`. If that's the case, you can move this down there so that it's closer to the code that uses it.